pkgs/applications/editors/vim/plugins/update.py

```diff
@@ -1,181 +1,213 @@
+
+    repo: GHRepoInfo = field(
+        converter=lambda x: GHRepoInfo(**x) if not isinstance(x, GHRepoInfo) else x
+    )
+    branch: str | None
+    alias: str | None
+
+    @property
+    @t.override
+    def id(self) -> str:
+        id = self.alias or self.repo.name  # noqa: A001
+        return id.strip().replace(".", "-")
+
+    @t.override
+    async def fetch(self) -> VimEntry:
+        """Fetch plugin information from GitHub.
+
+        Returns:
+            VimEntry containing fetched repository and prefetch information
+        """
+        logger.debug(f"Fetching from GitHub {self.repo.owner}/{self.repo.name}")
+        github_token = os.environ.get("GITHUB_TOKEN")
+
+        if github_token is not None:
+            result = await github_fetch_graphql(
+                self.repo.owner, self.repo.name, github_token
             )
-            for pdesc, plugin in plugins:
-                content = self.plugin2nix(pdesc, plugin, _isNeovimPlugin(plugin))
-                f.write(content)
-                if (
-                    plugin.name == "nvim-treesitter"
-                    and plugin.commit != nvim_treesitter_rev
-                ):
-                    self.nvim_treesitter_updated = True
-            f.write("\n}\n")
-        print(f"updated {outfile}")
-
-    def plugin2nix(
-        self, pdesc: PluginDesc, plugin: pluginupdate.Plugin, isNeovim: bool
-    ) -> str:
-        if isNeovim:
-            raise RuntimeError(f"Plugin {plugin.name} is already packaged in `luaPackages`, please use that")
-        repo = pdesc.repo
-
-        content = f"  {plugin.normalized_name} = "
-        src_nix = repo.as_nix(plugin)
-        content += """{buildFn} {{
-    pname = "{plugin.name}";
-    version = "{plugin.version}";
-    src = {src_nix};
-    meta.homepage = "{repo.uri}";
-    meta.hydraPlatforms = [ ];
-  }};
-
-""".format(
-            buildFn="buildNeovimPlugin" if isNeovim else "buildVimPlugin",
-            plugin=plugin,
-            src_nix=src_nix,
-            repo=repo,
-        )
-        log.debug(content)
-        return content
-
-    def update(self, args):
-        pluginupdate.update_plugins(self, args)
-
-        # TODO this should probably be skipped when running outside a nixpkgs checkout
-        if self.nvim_treesitter_updated:
-            print("updating nvim-treesitter grammars")
-            cmd = [
-                "nix",
-                "build",
-                "vimPlugins.nvim-treesitter.src",
-                "-f",
-                self.nixpkgs,
-                "--print-out-paths",
-            ]
-            log.debug("Running command: %s", " ".join(cmd))
-            nvim_treesitter_dir = subprocess.check_output(
-                cmd, text=True, timeout=90
-            ).strip()
-
-            generated = treesitter.update_grammars(nvim_treesitter_dir)
-            treesitter_generated_nix_path = os.path.join(
-                NIXPKGS_NVIMTREESITTER_FOLDER, "generated.nix"
```

Comment:
Is there a reason we removed all the logic around treesitter updates?

Reply:
Yes, too hard to implement currently. I will make a separate PR which makes treesitter updates separate from the general updater

Reply:
@khaneliman Does this block this PR? If so maybe an interim solution is needed and/or the follow up PR should be worked on before this is merged?

One option may be tweaking the old updater to spit out the new `generated.json` format (in a smaller PR). That way this impl and the old impl could be used interchangeably for a while.

Or is the issue that the new `generated.json` file itself doesn't know how to represent treesitter grammars? They could always be moved to their own json file if that's simpler.